pkg/webhooks/ingresscontroller/ingresscontroller.go

```diff
@@ -149,12 +176,151 @@ func (wh *IngressControllerWebhook) authorized(request admissionctl.Request) adm
 		}
 	}
 
-	ret = admissionctl.Allowed("IngressController operation is allowed")
+	/* TODO:
+	 * 1) ONLY check for privatelink clusters? How?
+	 * 2) HCP vs Classic, etc.. ...any other cluster type this should apply to? Is this handled by the
+	 * HypershiftEnabled vs ClassicEnabled flags set in this module? Currently HCP disabled.
+	 * If HCP is to be enabled for allowed source ranges, should this part of a 2nd ingress validator to
+	 * allow separation of validations between cluster install types? Is there a run time method available
+	 * to this validator to determine classic vs hcp?
+	 * 3) What other specifics should be checked here for this cidr check to be applicable?m
+	 */
+	// Only check for machine cidr in allowed ranges if creating or updating resource...
+	reqOp := request.AdmissionRequest.Operation
+	if reqOp == admissionv1.Create || reqOp == admissionv1.Update {
+		//TODO: Will these need to iterate over more than just the default IngressController config?
```

Comment:
Only the `default` Ingress Controller is needed

Reply:
Lets start by just setting up guardrails around the `default` ingresscontroller, and if it turns out we need to check additional ICs, we can iterate.

There's little risk to this approach, IMO, since the changes here are adding new checks, so worst case scenario would be that we just don't reduce alerts as much as we expect.

Reply:
Thanks. Will leave the scope limited to the 'default' IC for now.